logstash_async/memory_cache.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# This software may be modified and distributed under the terms
+# of the MIT license.  See the LICENSE file for details.
+
+import uuid
+from datetime import datetime, timedelta
+from logstash_async.cache import Cache
+
+
+class MemoryCache(Cache):
+    """Backend implementation for python-logstash-async. Keeps messages in a local, in-memory cache
+    while attempting to publish them to logstash. Does not persist through process restarts. Also,
+    does not write to disk.
+
+    :param cache: Usually just an empty dictionary
+    :param event_ttl: Optional parameter used to expire events in the cache after a time
+    """
+
+    def __init__(self, cache, event_ttl=None):
+        self._cache = cache
+        self._event_ttl = event_ttl
+
+    def add_event(self, event):
+        event_id = uuid.uuid4()
+        self._cache[event_id] = {
+            "event_text": event,
+            "pending_delete": False,
+            "entry_date": datetime.now(),
+            "id": event_id
+        }
+
+    def get_queued_events(self):
+        events = []
+        for event in self._cache.values():
+            if not event['pending_delete']:
+                events.append(event)
+        return events
+
+    def requeue_queued_events(self, events):
+        for event in events:
+            self._cache[event['id']]['pending_delete'] = False
+
+    def delete_queued_events(self):
+        ids_to_delete = [event['id'] for event in self._cache.values() if event['pending_delete']]
+        self._delete_events(ids_to_delete)
+
+    def expire_events(self):
+        if self._event_ttl is None:
+            return
+
+        delete_time = datetime.now() - timedelta(seconds=self._event_ttl)
+        ids_to_delete = [event['id'] for event in self._cache.values() if event['entry_date'] < delete_time]
+        self._delete_events(ids_to_delete)
+
+    def _delete_events(self, ids_to_delete):
+        for event_id in ids_to_delete:
+            self._cache.pop(event_id)
```

Comment:
Similar as for `requeue_queued_events` maybe catch `KeyError` of the key is not in the cache or provide a default to `.pop()`.